java/libs/src/main/java/org/physical_web/collection/PhysicalWebCollection.java

```diff
@@ -260,6 +271,38 @@ public void jsonDeserialize(JSONObject jsonObject) throws PhysicalWebCollectionE
   }
 
   /**
+   * Return a list of UrlGroups sorted by the rank of the top-ranked pair in each group.
+   * @return a sorted list of UrlGroups.
+   */
+  public List<UrlGroup> getUrlGroupsSortedByRank() {
+    List<PwPair> sortedPwPairs = getPwPairsSortedByRank();
+
+    Map<String, UrlGroup> pwGroupMap = new HashMap<>();
+    for (PwPair pwPair : sortedPwPairs) {
+      PwsResult pwsResult = pwPair.getPwsResult();
+      if (pwsResult != null) {
+        String groupId = pwsResult.getGroupId();
+        if (groupId != null && !groupId.equals("")) {
```

Comment:
Assuming PWS hasn't left them out for some reason, no...we should remove this check too.